drivers/gpu/drm/rp1/rp1-dsi/rp1_dsi_dsi.c

```diff
@@ -1290,23 +1300,25 @@ static unsigned long rp1dsi_refclk_freq(struct rp1_dsi *dsi)
 	return u;
 }
 
-static void rp1dsi_dpiclk_start(struct rp1_dsi *dsi, unsigned int bpp, unsigned int lanes)
+static void rp1dsi_dpiclk_start(struct rp1_dsi *dsi, u32 byte_clock,
+				unsigned int bpp, unsigned int lanes)
 {
-	unsigned long u;
-
-	if (dsi->clocks[RP1DSI_CLOCK_DPI]) {
-		u = (dsi->clocks[RP1DSI_CLOCK_BYTE]) ?
-				clk_get_rate(dsi->clocks[RP1DSI_CLOCK_BYTE]) : 0;
-		drm_info(dsi->drm,
-			 "rp1dsi: Nominal byte clock %lu; scale by %u/%u",
-			 u, 4 * lanes, (bpp >> 1));
-		if (u < 1 || u >= (1ul << 28))
-			u = 72000000ul; /* default DUMMY frequency for byteclock */
-
-		clk_set_parent(dsi->clocks[RP1DSI_CLOCK_DPI], dsi->clocks[RP1DSI_CLOCK_BYTE]);
-		clk_set_rate(dsi->clocks[RP1DSI_CLOCK_DPI], (4 * lanes * u) / (bpp >> 1));
-		clk_prepare_enable(dsi->clocks[RP1DSI_CLOCK_DPI]);
-	}
+	if (!dsi->clocks[RP1DSI_CLOCK_DPI])
+		return;
+
+	if (bpp >= 8 * lanes && dsi->clocks[RP1DSI_CLOCK_BYTE] &&
+	    clk_set_parent(dsi->clocks[RP1DSI_CLOCK_DPI], dsi->clocks[RP1DSI_CLOCK_BYTE]) == 0)
+		byte_clock = clk_get_rate(dsi->clocks[RP1DSI_CLOCK_BYTE]); /* dummy/nominal rate */
+	else if (dsi->clocks[RP1DSI_CLOCK_PLLSYS])
+		clk_set_parent(dsi->clocks[RP1DSI_CLOCK_DPI], dsi->clocks[RP1DSI_CLOCK_PLLSYS]);
```

Comment:
You checked the return code above, but not here.

Reply:
Yes, not really sure what to do if they both fail, as it leaves us with an unknown clock parent (which might still work provided it has never been the dummy one).

Reply:
Yes, we're in void returning functions, so you are pretty stuffed. I was more commenting on the lack of consistency.

Doing a `git grep clk_set_parent` I see a number of drivers not checking the return value at all, so it's probably OK not to.

Reply:
Now checking neither of them.